Gfm testing report

egeerardyn opened the pull request:
This is a partial implementation of #675. Most of it is done, it's just a few minor things that I'd still like to address. But this allows me to get some feedback from Travis and from you guys whether you think this is an improvement. 

What is shown is the following:
- everything unreliable (for time to infinity, that overview should disappear)
- the reliable tests that fail or skip
- a summary table that breaks down passing/failing/skipping tests versus reliability

# TODO

 - [x] Also show an overview of what tests have been run in easily copiable format or even a small MATLAB fragment to allow reproducing the test run.
 - [x] Maybe also mention the environment/OS closer to the summary table
 - [x] Also include commit/m2t version/... information

Let's keep the clipboard stuff for another time.

**Formatted test report is below**

-----

# MATLAB 8.4 (Mac) commit 8fe78a4765 

## Unreliable tests
These do not cause the build to fail.

|   Testcase |                   Name |                       OK |                                                                      Status |
| :--------- | :--------------------- | :----------------------: | :-------------------------------------------------------------------------- |
|  `ACID(1)` |     `multiline_labels` | :heavy_exclamation_mark: | hash a2fb844c26e59f8089b54741a49e3d52 != (01d4eeae27254d91a0f8316897e23e81) |
|  `ACID(6)` | `sine_with_annotation` | :heavy_exclamation_mark: | hash bd96c5a63e3d67dcede593e8063cbe83 != (7d415c014eec35c249802eb3a3ee519e) |
| `ACID(10)` |       `peaks_contourf` | :heavy_exclamation_mark: | hash 09bb0e7b55bbcd64c5588f0b74746070 != (77bfa3a9f07e49b9061d919551ad0a0f) |
| `ACID(16)` |              `logplot` | :heavy_exclamation_mark: | hash 75608a2f050cec6bd699ae2f1d2aba3e != (8d0a305ae7092f6f4141d33a5902ee47) |
| `ACID(18)` |           `legendplot` | :heavy_exclamation_mark: | hash 8df5189ea4afc8c47d5a53189dc7e104 != (746423d1c0d403421667221a6b0528bf) |
| `ACID(24)` |          `quiver3plot` | :heavy_exclamation_mark: | hash 5bab28f8aab62ed68f729d17fed12d24 != (612ab8ee8ed66a0368767b39854bb274) |
| `ACID(34)` |                 `bars` | :heavy_exclamation_mark: | hash 54625f5dfc2309fa9ad69564dce80c29 != (26cfa565f6c7581d8a0820aa15946f0c) |
| `ACID(38)` |          `subplot2x2b` | :heavy_exclamation_mark: | hash 269a5b987a97fe321b1303a522795aa4 != (63cf7ce7f29fef045cc5ab3042c3e89b) |
| `ACID(40)` |        `subplotCustom` | :heavy_exclamation_mark: | hash 1d7fe17273f430278b5ba4b559ec9f6e != (c3ac3cf225706a3a157a7543c4b72c24) |
| `ACID(42)` |            `bodeplots` | :heavy_exclamation_mark: | hash 792303b7270ada312d132bbcded01bd8 != (8f6cafc1771cc3a506c8fc46c15ec43e) |
| `ACID(43)` |           `rlocusPlot` | :heavy_exclamation_mark: | hash 3755c998703a9a1453c1e5433830b3fa != (f372b9a4004fa45ac96a21014656e143) |
| `ACID(48)` |          `zplanePlot2` | :heavy_exclamation_mark: | hash b17979195d439feeff04884322b58db9 != (fb3ba1a4db46c9975cd00e6da99ebe48) |
| `ACID(58)` |            `surfPlot2` | :heavy_exclamation_mark: | hash 80a4bec384ed6d3fffb85fb268025315 != (eb8e6d18db994ba4a90e177bd44b5db3) |
| `ACID(94)` | `stackedBarsWithOther` | :heavy_exclamation_mark: | hash ef1a39444e514591464725da902ac073 != (979b797068fb431c57e85e813308a2ea) |
| `ACID(97)` |     `overlappingPlots` | :heavy_exclamation_mark: | hash 0fb042829e28c31e6cd96910974ea970 != (e3f4a07b2e345edb1d3ccfcafbe3d163) |

## Reliable tests
Only the reliable tests determine the build outcome.
Passing tests are not shown (only failed and skipped tests).

|   Testcase |              Name |              OK |  Status |
| :--------- | :---------------- | :-------------: | :------ |
| `ACID(17)` | `colorbarLogplot` | :grey_question: | SKIPPED |

## Test summary
Test results for m2t commit 8fe78a4765 running with MATLAB 8.4 on Mac.

```matlab
suite = @ACID;
alltests = [1:99];
reliable = [2:5 7:9 11:15 17 19:23 25:33 35:37 39 41 44:47 49:57 59:93 95:96 98:99];
unreliable = [1 6 10 16 18 24 34 38 40 42:43 48 58 94 97];
failReliable = [];
passUnreliable = [];
skipped = [17];
```

|            | Pass | Fail | Skip | Total |
| :--------- | ---: | ---: | ---: | ----: |
|   Reliable |   83 |    0 |    1 |    84 |
| Unreliable |    0 |   15 |    0 |    15 |
|      Total |   83 |   15 |    1 |    99 |

Build passes. :heavy_check_mark:

PeterPablo commented on test/makeTravisReport.m:
```diff
@@ -1,45 +1,260 @@
-    for iTest = 1:numel(status)
-        fprintf(stream, '%s\n', formatSummaryRow(status{iTest}));
+% == INPUT VALIDATOR FUNCTIONS =================================================
+function bool = isstream(val)
```
camelCase?

PeterPablo commented on test/makeTravisReport.m:
```diff
@@ -1,45 +1,260 @@
+    % returns true if it is a valid (file) stream or stdout/stderr stream
+    bool = isnumeric(val);
+end
+function bool = islength(val)
```
camelCase?

PeterPablo commented on test/makeTravisReport.m:
```diff
@@ -1,45 +1,260 @@
+        alignment = 'l';
+    end
+    if numel(alignment) < nCols
+        % repeat the alignment specifications to along the columns
```
`to align the`

PeterPablo commented on test/makeTravisReport.m:
```diff
@@ -1,45 +1,260 @@
+        strs{iRow} = sprintf(rowformat, thisRow{:});
+    end
+    str = [strs{:}];
+    function alignRow = formatAlignment(alignment, columnWidth)
```
should some kind of delimeter line be added between inline functions (also above and below)?

egeerardyn replied:
We can indeed do that.

PeterPablo commented on test/makeTravisReport.m:
```diff
@@ -1,45 +1,260 @@
+                     'skip', ':grey_question:');
+end
+% ==============================================================================
+function S = splitStatusses(status)
```
[`splitStatuses`](https://en.wiktionary.org/wiki/statuses#English)

PeterPablo commented on test/makeTravisReport.m:
```diff
@@ -1,45 +1,260 @@
+                    VCID, env, ver, getOS);
+    % maybe we need a bit more in the long format?
+end
+function OS = getOS
```
`getOS()`

egeerardyn replied:
I will remove `getOS`, since some forgetful guy wrote `OSVersion` which gives more information.

PeterPablo commented on test/makeTravisReport.m:
```diff
@@ -1,45 +1,260 @@
+    short = sprintf('%s %s (%s)', env, ver, getOS, VCID);
+    long  = sprintf('Test results for m2t%s running with %s %s on %s.', ...
+                    VCID, env, ver, getOS);
+    % maybe we need a bit more in the long format?
```
I'd remove the comment.

PeterPablo commented on test/makeTravisReport.m:
```diff
@@ -1,45 +1,260 @@
+end
+function OS = getOS
+    % Quick and dirty way to determine the OS
+    % Probably this will be wrong on Solaris (if somebody still uses that)
```
I'd remove the comment.

PeterPablo commented on test/makeTravisReport.m:
```diff
@@ -61,13 +276,91 @@ function displaySummaryTable(stream, status)
+        str = intelligentVector( cellfun(@(s) s.index, status) );
+    end
+end
+function str = intelligentVector(numbers)
```
well done!

egeerardyn replied:
Thanks :smile:

PeterPablo commented on test/makeTravisReport.m:
```diff
@@ -61,13 +276,91 @@ function displaySummaryTable(stream, status)
+    end
+end
+function str = formatRange(start, stop)
+    % format a range [start,stop] of integers in MATLAB syntax
```
`[start:stop]`

egeerardyn replied:
:+1:

PeterPablo commented on test/private/hasTestFailed.m:
```diff
@@ -0,0 +1,13 @@
+            bool = bool || status.(stages{jStage}).error;
+        end
+    end
+end
```
missing newline

PeterPablo commented on test/private/hasTestFailed.m:
```diff
@@ -0,0 +1,13 @@
+function bool = hasTestFailed(status)
```
Why is this function externalized to a private function?

egeerardyn replied:
Because this allows to call it from `countNumberOfErrors`, which is simplified quite a bit this way.

PeterPablo replied:
agreed!

PeterPablo commented on test/private/splitPassFailSkippedTests.m:
```diff
@@ -0,0 +1,10 @@
+function [passedTests, failedTests, skippedTests] = splitPassFailSkippedTests(status)
```
Why is this function externalized to a private function?

egeerardyn replied:
This could be used in the LaTeX report as well.

PeterPablo replied:
:+1:

PeterPablo commented:
Impressive and very nice looking piece of code! I haven't yet given it a spin, but it looks good to me! Since this question might re-arise and I didn't notice it while looking at the code: How does the developer specify the log level?

PeterPablo commented on test/makeTravisReport.m:
```diff
@@ -1,45 +1,260 @@
+end
+function symbols = githubEmoji()
+    % defines the emojis to signal the test result
+    symbols = struct('pass', ':heavy_check_mark:', ...
```
Could we use `:white_check_mark: ` :white_check_mark:  to make the color transport the message as well?

egeerardyn replied:
Surely.

PeterPablo commented:
`status = testHeadless(); makeTravisReport(status);`

yields
```
# Octave 3.8.2 (Mac) commit 2118e466e2 

## Unreliable tests
These do not cause the build to fail.

|   Testcase |                      Name |                       OK |                                                                      Status |
| :--------- | :------------------------ | :----------------------: | :-------------------------------------------------------------------------- |
|  `ACID(1)` |        `multiline_labels` | :heavy_exclamation_mark: | hash f32981ed7cfd6f15bb98251d1d793540 != (9d4370a4384c4dcd804baec5c4b6af63) |
|  `ACID(6)` |    `sine_with_annotation` | :heavy_exclamation_mark: | hash 0f3f77172ef7a8f91e3d1a666181a809 != (92e9da02d8ff8cfd7d851ddf29c73123) |
|  `ACID(8)` |           `peaks_contour` | :heavy_exclamation_mark: | hash 174c4474fa88a471a210f5993e395a5f != (7d1a67eb06d87860adca396396e576cd) |
| `ACID(17)` |         `colorbarLogplot` |       :heavy_check_mark: |                                                                             |
| `ACID(18)` |              `legendplot` | :heavy_exclamation_mark: | hash 503c6ecf51c6a3b6d8c0794faf4b4bc7 != (45f5eae108b3d13a611561ec6a842922) |
| `ACID(21)` |                    `zoom` | :heavy_exclamation_mark: | hash 68938879b9798f9af2091243f95ac0d9 != (0763cf25f523e713247c03bdb398615f) |
| `ACID(25)` |            `logicalImage` |       :heavy_check_mark: |                                                                             |
| `ACID(26)` |             `imagescplot` |       :heavy_check_mark: |                                                                             |
| `ACID(27)` |            `imagescplot2` |       :heavy_check_mark: |                                                                             |
| `ACID(34)` |                    `bars` | :heavy_exclamation_mark: | hash 28abac3723ccc0f7e1477b193be804ce != (3b7ac8767e299b849c3cbdeee804f0c5) |
| `ACID(38)` |             `subplot2x2b` | :heavy_exclamation_mark: | hash b2861693605d5ac287b615570f45ee98 != (097ecc9418fb5826340f3903fed8a49f) |
| `ACID(41)` |          `legendsubplots` | :heavy_exclamation_mark: | hash 0c284c3a83f9ace2ec368d69386da29e != (5005469bf8cbe365f00ef9c78d955c1b) |
| `ACID(45)` |             `besselImage` |       :heavy_check_mark: |                                                                             |
| `ACID(58)` |               `surfPlot2` | :heavy_exclamation_mark: | hash 4cdf6e4ede9b004f3e018fff254bba97 != (9e9f49e3f6e08837538ab6e05ec43e90) |
| `ACID(63)` |            `mixedBarLine` |       :heavy_check_mark: |                                                                             |
| `ACID(67)` |               `texrandom` | :heavy_exclamation_mark: | hash 2ca0c7527f12fd26081e650d3d56e35c != (d3be1866d7ec69725c713c35b45a0d34) |
| `ACID(78)` |            `customLegend` |       :heavy_check_mark: |                                                                             |
| `ACID(86)` |              `alphaImage` | :heavy_exclamation_mark: | hash 6e5aa7efd6ead5bfe6f7fddae0425360 != (cbb2cf8ab5ccd1c4d27bb40cf8617214) |
| `ACID(91)` |    `imageOrientation_PNG` | :heavy_exclamation_mark: | hash 81779a0fc475e03d90935188ed75c222 != (9e4a63ccad80d0842cdb6a0fe20cc0ad) |
| `ACID(92)` | `imageOrientation_inline` | :heavy_exclamation_mark: | hash 9fbbe6d118517816b700e74b66d630ee != (365a0d1f57f0626c2020c3f719e2c553) |
| `ACID(94)` |    `stackedBarsWithOther` | :heavy_exclamation_mark: | hash adefc2773afff16dc1315e43c0d02450 != (e3cd431e967814dfd0c6f92e8f4de87f) |
| `ACID(95)` |      `colorbarLabelTitle` | :heavy_exclamation_mark: | hash 8a0c9a6b48750ffde7eda66ff2216e4d != (d3c0aa0e9481c5ace5d64eeb50932e38) |
| `ACID(96)` |           `textAlignment` | :heavy_exclamation_mark: | hash 466ac44f3372736978518f1c6a3c2bc2 != (ab67c21b9a17cb4fc55cf2b112977581) |
error: horizontal dimensions mismatch (1x1 vs 23x1)
error: called from:
error:   /Users/mypath/0_addons/figures/matlab2tikz/test/makeTravisReport.m at line 227, column 19
error:   /Users/mypath/0_addons/figures/matlab2tikz/test/makeTravisReport.m at line 38, column 5
```

PeterPablo commented:
Take a look at my octave 4.0 branch. I had the same issue there resolved... Probably there is a struct `s` that needs to be "prepared" by `s = s(:)` or similar.

PeterPablo commented:
R2014b
```
>> status = testHeadless(42); nErrors = makeTravisReport(status,'length','verbose');
Initialize global state...
Initialize working directory...
Successfully changed into test folder...
Executing ACID test no. 42...
bodeplots done (2.00s).

overall time: 2.00s

Restore global state...
Restore working directory...
Error using m2tInputParser>parse (line 180)
m2tInputParser: Input argument 'length' has invalid value.


Error in makeTravisReport (line 28)
    ipp = ipp.parse(ipp, status, varargin{:});
 ```

PeterPablo commented on test/makeTravisReport.m:
```diff
@@ -61,13 +276,91 @@ function displaySummaryTable(stream, status)
+    summary = arrayfun(@(v) sprintf('%d',v), summary, 'UniformOutput', false);
+    table = repmat({''}, 3, 5);
+    header = {'','Pass','Fail','Skip','Total'};
+    table(:,1) = {'Reliable','Unreliable','Total'};
```
Should we change this to "unreliable, reliable, total" (different order), as it used to be?

egeerardyn commented:
I think I have addressed all your comments. I still cannot check Octave 4 (won't compile on my Arch box), but I now paid some attention to the cell concatenation dimensions.

Current output is below

---------

# Octave 3.8.0 (Mac OS 10.10.4) commit b4e2cca28f 

## Unreliable tests
These do not cause the build to fail.

|   Testcase |                      Name |                       OK |                                                                      Status |
| :--------- | :------------------------ | :----------------------: | :-------------------------------------------------------------------------- |
|  `ACID(1)` |        `multiline_labels` | :heavy_exclamation_mark: | hash f32981ed7cfd6f15bb98251d1d793540 != (9d4370a4384c4dcd804baec5c4b6af63) |
|  `ACID(6)` |    `sine_with_annotation` | :heavy_exclamation_mark: | hash 0f3f77172ef7a8f91e3d1a666181a809 != (92e9da02d8ff8cfd7d851ddf29c73123) |
|  `ACID(8)` |           `peaks_contour` | :heavy_exclamation_mark: | hash 10f2b9bc1b7f88583a5ec6dac1954d79 != (7d1a67eb06d87860adca396396e576cd) |
| `ACID(17)` |         `colorbarLogplot` |       :white_check_mark: |                                                                             |
| `ACID(18)` |              `legendplot` | :heavy_exclamation_mark: | hash 503c6ecf51c6a3b6d8c0794faf4b4bc7 != (45f5eae108b3d13a611561ec6a842922) |
| `ACID(21)` |                    `zoom` | :heavy_exclamation_mark: | hash 68938879b9798f9af2091243f95ac0d9 != (0763cf25f523e713247c03bdb398615f) |
| `ACID(25)` |            `logicalImage` |       :white_check_mark: |                                                                             |
| `ACID(26)` |             `imagescplot` |       :white_check_mark: |                                                                             |
| `ACID(27)` |            `imagescplot2` |       :white_check_mark: |                                                                             |
| `ACID(34)` |                    `bars` | :heavy_exclamation_mark: | hash 28abac3723ccc0f7e1477b193be804ce != (3b7ac8767e299b849c3cbdeee804f0c5) |
| `ACID(38)` |             `subplot2x2b` | :heavy_exclamation_mark: | hash b2861693605d5ac287b615570f45ee98 != (097ecc9418fb5826340f3903fed8a49f) |
| `ACID(41)` |          `legendsubplots` | :heavy_exclamation_mark: | hash 0c284c3a83f9ace2ec368d69386da29e != (5005469bf8cbe365f00ef9c78d955c1b) |
| `ACID(45)` |             `besselImage` |       :white_check_mark: |                                                                             |
| `ACID(58)` |               `surfPlot2` | :heavy_exclamation_mark: | hash 4cdf6e4ede9b004f3e018fff254bba97 != (9e9f49e3f6e08837538ab6e05ec43e90) |
| `ACID(63)` |            `mixedBarLine` | :heavy_exclamation_mark: | hash ddd1440395b1249570205c5a6affc2e1 != (d104e14ad08f82223ef6e7bfb0251fcd) |
| `ACID(67)` |               `texrandom` | :heavy_exclamation_mark: | hash fbcd8022b17c28e78928bf4dbaaab713 != (d3be1866d7ec69725c713c35b45a0d34) |
| `ACID(78)` |            `customLegend` |       :white_check_mark: |                                                                             |
| `ACID(86)` |              `alphaImage` | :heavy_exclamation_mark: | hash 6e5aa7efd6ead5bfe6f7fddae0425360 != (cbb2cf8ab5ccd1c4d27bb40cf8617214) |
| `ACID(91)` |    `imageOrientation_PNG` | :heavy_exclamation_mark: | hash 81779a0fc475e03d90935188ed75c222 != (9e4a63ccad80d0842cdb6a0fe20cc0ad) |
| `ACID(92)` | `imageOrientation_inline` | :heavy_exclamation_mark: | hash 9fbbe6d118517816b700e74b66d630ee != (365a0d1f57f0626c2020c3f719e2c553) |
| `ACID(94)` |    `stackedBarsWithOther` | :heavy_exclamation_mark: | hash adefc2773afff16dc1315e43c0d02450 != (e3cd431e967814dfd0c6f92e8f4de87f) |
| `ACID(95)` |      `colorbarLabelTitle` | :heavy_exclamation_mark: | hash 8a0c9a6b48750ffde7eda66ff2216e4d != (d3c0aa0e9481c5ace5d64eeb50932e38) |
| `ACID(96)` |           `textAlignment` | :heavy_exclamation_mark: | hash 000b946a520c4fd0fd0feef6e54427e1 != (ab67c21b9a17cb4fc55cf2b112977581) |

## Reliable tests
Only the reliable tests determine the build outcome.
Passing tests are not shown (only failed and skipped tests).

|   Testcase |                  Name |              OK |  Status |
| :--------- | :-------------------- | :-------------: | :------ |
|  `ACID(9)` |        `contourPenny` | :grey_question: | SKIPPED |
| `ACID(12)` |     `double_colorbar` | :grey_question: | SKIPPED |
| `ACID(42)` |           `bodeplots` | :grey_question: | SKIPPED |
| `ACID(43)` |          `rlocusPlot` | :grey_question: | SKIPPED |
| `ACID(44)` |       `mandrillImage` | :grey_question: | SKIPPED |
| `ACID(46)` |          `clownImage` | :grey_question: | SKIPPED |
| `ACID(47)` |         `zplanePlot1` | :grey_question: | SKIPPED |
| `ACID(48)` |         `zplanePlot2` | :grey_question: | SKIPPED |
| `ACID(49)` |    `freqResponsePlot` | :grey_question: | SKIPPED |
| `ACID(54)` |         `scatterPlot` | :grey_question: | SKIPPED |
| `ACID(59)` |          `superkohle` | :grey_question: | SKIPPED |
| `ACID(62)` |             `spectro` | :grey_question: | SKIPPED |
| `ACID(71)` |       `parameterSurf` | :grey_question: | SKIPPED |
| `ACID(72)` |           `fill3plot` | :grey_question: | SKIPPED |
| `ACID(75)` |              `hist3d` | :grey_question: | SKIPPED |
| `ACID(76)` |           `myBoxplot` | :grey_question: | SKIPPED |
| `ACID(80)` |        `croppedImage` | :grey_question: | SKIPPED |
| `ACID(82)` |     `hgTransformPlot` | :grey_question: | SKIPPED |
| `ACID(87)` |       `annotationAll` | :grey_question: | SKIPPED |
| `ACID(88)` |  `annotationSubplots` | :grey_question: | SKIPPED |
| `ACID(89)` |      `annotationText` | :grey_question: | SKIPPED |
| `ACID(90)` | `annotationTextUnits` | :grey_question: | SKIPPED |
| `ACID(98)` |       `histogramPlot` | :grey_question: | SKIPPED |

## Test summary
Test results for m2t commit b4e2cca28f running with Octave 3.8.0 on Mac OS 10.10.4.

```matlab
suite = @ACID;
alltests = [1:99];
reliable = [2:5 7 9:16 19:20 22:24 28:33 35:37 39:40 42:44 46:57 59:62 64:66 68:77 79:85 87:90 93 97:99];
unreliable = [1 6 8 17:18 21 25:27 34 38 41 45 58 63 67 78 86 91:92 94:96];
failReliable = [];
passUnreliable = [17 25:27 45 78];
skipped = [9 12 42:44 46:49 54 59 62 71:72 75:76 80 82 87:90 98];
```

|            | Pass | Fail | Skip | Total |
| :--------- | ---: | ---: | ---: | ----: |
| Unreliable |    6 |   17 |    0 |    23 |
|   Reliable |   53 |    0 |   23 |    76 |
|      Total |   59 |   17 |   23 |    99 |

Build passes. :white_check_mark:

PeterPablo commented:
On octave 3.8.2 (Mac):
```
error: horizontal dimensions mismatch (1x1 vs 23x1)
error: called from:
error:   /Users/mypath/0_addons/figures/matlab2tikz/test/makeTravisReport.m at line 227, column 19
error:   /Users/mypath/0_addons/figures/matlab2tikz/test/makeTravisReport.m at line 38, column 5
```

egeerardyn commented:
@PeterPablo Can you double-check that, because at b4e2cca I have the following on line 227:
```matlab
    fprintf(arg.stream, gfmHeader('Reliable tests',2));
```
i.e. no concatenation happening.

Running on my Octave, I don't get any problems. If you can still reproduce, would you be so kind to save your `status` variable from `testHeadless` to a file such that I can try to reproduce?

PeterPablo commented:
```
# Octave 3.8.2 (Mac) commit b4e2cca28f 

## Unreliable tests
These do not cause the build to fail.

|   Testcase |                      Name |                       OK |                                                                      Status |
| :--------- | :------------------------ | :----------------------: | :-------------------------------------------------------------------------- |
|  `ACID(1)` |        `multiline_labels` | :heavy_exclamation_mark: | hash f32981ed7cfd6f15bb98251d1d793540 != (9d4370a4384c4dcd804baec5c4b6af63) |
|  `ACID(6)` |    `sine_with_annotation` | :heavy_exclamation_mark: | hash 0f3f77172ef7a8f91e3d1a666181a809 != (92e9da02d8ff8cfd7d851ddf29c73123) |
|  `ACID(8)` |           `peaks_contour` | :heavy_exclamation_mark: | hash 174c4474fa88a471a210f5993e395a5f != (7d1a67eb06d87860adca396396e576cd) |
| `ACID(17)` |         `colorbarLogplot` |       :heavy_check_mark: |                                                                             |
| `ACID(18)` |              `legendplot` | :heavy_exclamation_mark: | hash 503c6ecf51c6a3b6d8c0794faf4b4bc7 != (45f5eae108b3d13a611561ec6a842922) |
| `ACID(21)` |                    `zoom` | :heavy_exclamation_mark: | hash 68938879b9798f9af2091243f95ac0d9 != (0763cf25f523e713247c03bdb398615f) |
| `ACID(25)` |            `logicalImage` |       :heavy_check_mark: |                                                                             |
| `ACID(26)` |             `imagescplot` |       :heavy_check_mark: |                                                                             |
| `ACID(27)` |            `imagescplot2` |       :heavy_check_mark: |                                                                             |
| `ACID(34)` |                    `bars` | :heavy_exclamation_mark: | hash 28abac3723ccc0f7e1477b193be804ce != (3b7ac8767e299b849c3cbdeee804f0c5) |
| `ACID(38)` |             `subplot2x2b` | :heavy_exclamation_mark: | hash b2861693605d5ac287b615570f45ee98 != (097ecc9418fb5826340f3903fed8a49f) |
| `ACID(41)` |          `legendsubplots` | :heavy_exclamation_mark: | hash 0c284c3a83f9ace2ec368d69386da29e != (5005469bf8cbe365f00ef9c78d955c1b) |
| `ACID(45)` |             `besselImage` |       :heavy_check_mark: |                                                                             |
| `ACID(58)` |               `surfPlot2` | :heavy_exclamation_mark: | hash 4cdf6e4ede9b004f3e018fff254bba97 != (9e9f49e3f6e08837538ab6e05ec43e90) |
| `ACID(63)` |            `mixedBarLine` |       :heavy_check_mark: |                                                                             |
| `ACID(67)` |               `texrandom` | :heavy_exclamation_mark: | hash d95afc28191c88fb4940c60c49792732 != (d3be1866d7ec69725c713c35b45a0d34) |
| `ACID(78)` |            `customLegend` |       :heavy_check_mark: |                                                                             |
| `ACID(86)` |              `alphaImage` | :heavy_exclamation_mark: | hash 6e5aa7efd6ead5bfe6f7fddae0425360 != (cbb2cf8ab5ccd1c4d27bb40cf8617214) |
| `ACID(91)` |    `imageOrientation_PNG` | :heavy_exclamation_mark: | hash 81779a0fc475e03d90935188ed75c222 != (9e4a63ccad80d0842cdb6a0fe20cc0ad) |
| `ACID(92)` | `imageOrientation_inline` | :heavy_exclamation_mark: | hash 9fbbe6d118517816b700e74b66d630ee != (365a0d1f57f0626c2020c3f719e2c553) |
| `ACID(94)` |    `stackedBarsWithOther` | :heavy_exclamation_mark: | hash adefc2773afff16dc1315e43c0d02450 != (e3cd431e967814dfd0c6f92e8f4de87f) |
| `ACID(95)` |      `colorbarLabelTitle` | :heavy_exclamation_mark: | hash 8a0c9a6b48750ffde7eda66ff2216e4d != (d3c0aa0e9481c5ace5d64eeb50932e38) |
| `ACID(96)` |           `textAlignment` | :heavy_exclamation_mark: | hash 466ac44f3372736978518f1c6a3c2bc2 != (ab67c21b9a17cb4fc55cf2b112977581) |
error: horizontal dimensions mismatch (1x1 vs 23x1)
error: called from:
error:   /Users/path/0_addons/figures/matlab2tikz/test/makeTravisReport.m at line 227, column 19
error:   /Users/path/0_addons/figures/matlab2tikz/test/makeTravisReport.m at line 38, column 5
```

I will edit shortly

egeerardyn commented:
I get the following from your `status` variable (after patching up your path to match mine in the "MAT" file):

# Octave 3.8.0 (Mac OS 10.10.4) commit aa0c8cc383 

## Unreliable tests
These do not cause the build to fail.

|   Testcase |                      Name |                       OK |                                                                      Status |
| :--------- | :------------------------ | :----------------------: | :-------------------------------------------------------------------------- |
|  `ACID(1)` |        `multiline_labels` | :heavy_exclamation_mark: | hash f32981ed7cfd6f15bb98251d1d793540 != (9d4370a4384c4dcd804baec5c4b6af63) |
|  `ACID(6)` |    `sine_with_annotation` | :heavy_exclamation_mark: | hash 0f3f77172ef7a8f91e3d1a666181a809 != (92e9da02d8ff8cfd7d851ddf29c73123) |
|  `ACID(8)` |           `peaks_contour` | :heavy_exclamation_mark: | hash 174c4474fa88a471a210f5993e395a5f != (7d1a67eb06d87860adca396396e576cd) |
| `ACID(17)` |         `colorbarLogplot` |       :white_check_mark: |                                                                             |
| `ACID(18)` |              `legendplot` | :heavy_exclamation_mark: | hash 503c6ecf51c6a3b6d8c0794faf4b4bc7 != (45f5eae108b3d13a611561ec6a842922) |
| `ACID(21)` |                    `zoom` | :heavy_exclamation_mark: | hash 68938879b9798f9af2091243f95ac0d9 != (0763cf25f523e713247c03bdb398615f) |
| `ACID(25)` |            `logicalImage` |       :white_check_mark: |                                                                             |
| `ACID(26)` |             `imagescplot` |       :white_check_mark: |                                                                             |
| `ACID(27)` |            `imagescplot2` |       :white_check_mark: |                                                                             |
| `ACID(34)` |                    `bars` | :heavy_exclamation_mark: | hash 28abac3723ccc0f7e1477b193be804ce != (3b7ac8767e299b849c3cbdeee804f0c5) |
| `ACID(38)` |             `subplot2x2b` | :heavy_exclamation_mark: | hash b2861693605d5ac287b615570f45ee98 != (097ecc9418fb5826340f3903fed8a49f) |
| `ACID(41)` |          `legendsubplots` | :heavy_exclamation_mark: | hash 0c284c3a83f9ace2ec368d69386da29e != (5005469bf8cbe365f00ef9c78d955c1b) |
| `ACID(45)` |             `besselImage` |       :white_check_mark: |                                                                             |
| `ACID(58)` |               `surfPlot2` | :heavy_exclamation_mark: | hash 4cdf6e4ede9b004f3e018fff254bba97 != (9e9f49e3f6e08837538ab6e05ec43e90) |
| `ACID(63)` |            `mixedBarLine` |       :white_check_mark: |                                                                             |
| `ACID(67)` |               `texrandom` | :heavy_exclamation_mark: | hash bb6a432e068d5477feae183da0451542 != (d3be1866d7ec69725c713c35b45a0d34) |
| `ACID(78)` |            `customLegend` |       :white_check_mark: |                                                                             |
| `ACID(86)` |              `alphaImage` | :heavy_exclamation_mark: | hash 6e5aa7efd6ead5bfe6f7fddae0425360 != (cbb2cf8ab5ccd1c4d27bb40cf8617214) |
| `ACID(91)` |    `imageOrientation_PNG` | :heavy_exclamation_mark: | hash 81779a0fc475e03d90935188ed75c222 != (9e4a63ccad80d0842cdb6a0fe20cc0ad) |
| `ACID(92)` | `imageOrientation_inline` | :heavy_exclamation_mark: | hash 9fbbe6d118517816b700e74b66d630ee != (365a0d1f57f0626c2020c3f719e2c553) |
| `ACID(94)` |    `stackedBarsWithOther` | :heavy_exclamation_mark: | hash adefc2773afff16dc1315e43c0d02450 != (e3cd431e967814dfd0c6f92e8f4de87f) |
| `ACID(95)` |      `colorbarLabelTitle` | :heavy_exclamation_mark: | hash 8a0c9a6b48750ffde7eda66ff2216e4d != (d3c0aa0e9481c5ace5d64eeb50932e38) |
| `ACID(96)` |           `textAlignment` | :heavy_exclamation_mark: | hash 466ac44f3372736978518f1c6a3c2bc2 != (ab67c21b9a17cb4fc55cf2b112977581) |

## Reliable tests
Only the reliable tests determine the build outcome.
Passing tests are not shown (only failed and skipped tests).

|   Testcase |                  Name |                       OK |                                                                      Status |
| :--------- | :-------------------- | :----------------------: | :-------------------------------------------------------------------------- |
| `ACID(10)` |      `peaks_contourf` | :heavy_exclamation_mark: | hash 15452cb6280b7c00aa5e2b5ed5b2910b != (957fcc67221ce5c54b1981b2675348a1) |
|  `ACID(9)` |        `contourPenny` |          :grey_question: |                                                                     SKIPPED |
| `ACID(12)` |     `double_colorbar` |          :grey_question: |                                                                     SKIPPED |
| `ACID(42)` |           `bodeplots` |          :grey_question: |                                                                     SKIPPED |
| `ACID(43)` |          `rlocusPlot` |          :grey_question: |                                                                     SKIPPED |
| `ACID(44)` |       `mandrillImage` |          :grey_question: |                                                                     SKIPPED |
| `ACID(46)` |          `clownImage` |          :grey_question: |                                                                     SKIPPED |
| `ACID(47)` |         `zplanePlot1` |          :grey_question: |                                                                     SKIPPED |
| `ACID(48)` |         `zplanePlot2` |          :grey_question: |                                                                     SKIPPED |
| `ACID(49)` |    `freqResponsePlot` |          :grey_question: |                                                                     SKIPPED |
| `ACID(54)` |         `scatterPlot` |          :grey_question: |                                                                     SKIPPED |
| `ACID(59)` |          `superkohle` |          :grey_question: |                                                                     SKIPPED |
| `ACID(62)` |             `spectro` |          :grey_question: |                                                                     SKIPPED |
| `ACID(71)` |       `parameterSurf` |          :grey_question: |                                                                     SKIPPED |
| `ACID(72)` |           `fill3plot` |          :grey_question: |                                                                     SKIPPED |
| `ACID(75)` |              `hist3d` |          :grey_question: |                                                                     SKIPPED |
| `ACID(76)` |           `myBoxplot` |          :grey_question: |                                                                     SKIPPED |
| `ACID(80)` |        `croppedImage` |          :grey_question: |                                                                     SKIPPED |
| `ACID(82)` |     `hgTransformPlot` |          :grey_question: |                                                                     SKIPPED |
| `ACID(87)` |       `annotationAll` |          :grey_question: |                                                                     SKIPPED |
| `ACID(88)` |  `annotationSubplots` |          :grey_question: |                                                                     SKIPPED |
| `ACID(89)` |      `annotationText` |          :grey_question: |                                                                     SKIPPED |
| `ACID(90)` | `annotationTextUnits` |          :grey_question: |                                                                     SKIPPED |
| `ACID(98)` |       `histogramPlot` |          :grey_question: |                                                                     SKIPPED |

## Test summary
Test results for m2t commit aa0c8cc383 running with Octave 3.8.0 on Mac OS 10.10.4.

```matlab
suite = @ACID;
alltests = [1:99];
reliable = [2:5 7 9:16 19:20 22:24 28:33 35:37 39:40 42:44 46:57 59:62 64:66 68:77 79:85 87:90 93 97:99];
unreliable = [1 6 8 17:18 21 25:27 34 38 41 45 58 63 67 78 86 91:92 94:96];
failReliable = [10];
passUnreliable = [17 25:27 45 63 78];
skipped = [9 12 42:44 46:49 54 59 62 71:72 75:76 80 82 87:90 98];
```

|            | Pass | Fail | Skip | Total |
| :--------- | ---: | ---: | ---: | ----: |
| Unreliable |    7 |   16 |    0 |    23 |
|   Reliable |   52 |    1 |   23 |    76 |
|      Total |   59 |   17 |   23 |    99 |

Build fails with 1 errors. :heavy_exclamation_mark:

# Conclusion
This is a difference between Octave 3.8.2 and Octave 3.8.0.

PeterPablo commented:
Ok, I'll have a quick look at the code.

PeterPablo commented:
```diff
diff --git a/test/makeTravisReport.m b/test/makeTravisReport.m
index f26458c..279073e 100644
--- a/test/makeTravisReport.m
+++ b/test/makeTravisReport.m
@@ -224,7 +224,7 @@ function reportReliableTests(arg, S)
             tests = S.reliable;
             message = '';
         case 'default'
-            tests = [S.failR S.skipR];
+            tests = [S.failR; S.skipR];
             message = 'Passing tests are not shown (only failed and skipped tests).\n\n';
         case 'short'
             return; % don't show this part
@@ -319,6 +319,7 @@ function displayTestSummary(stream, S)
     if numel(S.failR) == 0
         fprintf(stream, '\nBuild passes. %s\n', symbol.pass);
     else
+        nErrors = numel(S.failR);
         fprintf(stream, '\nBuild fails with %d errors. %s\n', nErrors, symbol.fail);
     end
 end
```
The definition for nErrors of course can go up before the if-statement.

PeterPablo commented:
Apart from that I see no further issues. Tiny thing: Would you mind moving the MATLAB code block further up in the output. It is mostly obfuscating of what the dev is looking for: which tests failed (with names).

PeterPablo commented:
hmmm, you fixed the same things. For some reason my octave doesn't pick the new files it seems. Wierd!

egeerardyn commented:
@PeterPablo The last commit is in develop as I see it here and the code is also (judging by Sublime Text here locally).

What do you mean by moving the code block further up? Where would you like it? For me it's at the right spot there: in the summary. It allows to copy-paste that fragment and easily call the suite to reproduce the test run (e.g. only for failing tests). Another place that I could think of, is completely at the beginning such that in the `short` format, it still appears.

PeterPablo commented:
Yeah, I meant on the very beginning. But it is totally fine like this as well.
I must have had some weird GIT issue. I afterwards realized that you fixed the same two issues -- that's why I merged it.

egeerardyn commented:
I'll be on HipChat if you want to discuss that issue further.
